src/Compilers/Core/Portable/Symbols/IArrayTypeSymbol.cs

```diff
@@ -21,6 +21,28 @@ public interface IArrayTypeSymbol : ITypeSymbol
         int Rank { get; }
 
         /// <summary>
+        /// Is this a zero-based one-dimensional array, i.e. SZArray in CLR terms.
+        /// SZArray is an array type encoded in metadata with ELEMENT_TYPE_SZARRAY (always single-dim array with 0 lower bound).
+        /// Non-SZArray type is ecoded in metadata with ELEMENT_TYPE_ARRAY and with optional sizes and lower bounds. Even though 
+        /// non-SZArray can also be a single-dim array with 0 lower bound, the encoding of these types in metadata is distinct.
+        /// </summary>
+        bool IsSZArray { get; }
+
+        /// <summary>
+        /// Specified lower bounds for dimensions, by position. The length can be less than <see cref="Rank"/>,
+        /// meaning that some trailing dimensions don't have the lower bound specified.
+        /// The most common case is all dimensions are zero bound - a default (Nothing in VB) array is returned in this case.
+        /// </summary>
+        ImmutableArray<int> LowerBounds { get; }
+
+        /// <summary>
+        /// Specified sizes for dimensions, by position. The length can be less than <see cref="Rank"/>,
+        /// meaning that some trailing dimensions don't have the size specified.
+        /// The most common case is none of the dimensions have the size specified - an empty array is returned.
+        /// </summary>
+        ImmutableArray<int> Sizes { get; }
```

Comment:
If there are no explicit lower bounds, the LowerBounds API returns an empty array. The default array is returned for lower bounds when they are explicitly specified and all are 0. These are two distinct situations. The motivation is to decrease allocations in the most common scenario.